src/components/HomePage/SearchResultsArea/Offer/OfferVisibilityOptions.js

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+
+import { Grid, Tooltip, Divider, Button } from "@material-ui/core";
+import { Visibility, VisibilityOff, Block } from "@material-ui/icons";
+
+import { enableOffer, hideOffer } from "../../../../services/offerVisibilityService";
+
+const OfferVisibilityOptions = ({
+    loading, sessionData, classes, visibilityState,
+    setVisibilityState, addSnackbar, setShowAdminReasonModal,
+    dealWithPromiseError, isMobile, offer }) => {
+
+    const handleOfferVisibility = () => {
+        if (visibilityState.isVisible) {
+            hideOffer(offer.id).then(() => {
+                offer.isHidden = true;
```

Comment:
same thing as before, do not mutate offer directly, instead use a function that should be passed from above to mutate the offer object correctly